libs/fairness/src/lib/Controls/ModelComparisonChart.tsx

```diff
@@ -189,30 +231,94 @@ export class ModelComparisonChart extends React.Component<
         );
 
       const props = _.cloneDeep(this.plotlyProps);
+
+      const performanceMetricTitle = selectedMetric.title;
+      const fairnessMetric =
+        fairnessOptions[this.props.fairnessPickerProps.selectedFairnessKey];
+      const fairnessMetricTitle = fairnessMetric.title;
       props.data = ChartBuilder.buildPlotlySeries(props.data[0], data).map(
         (series) => {
           if (series.name) {
             series.name = this.props.dashboardContext.modelNames[series.name];
           }
+
+          series.customdata = [] as unknown as Datum[];
+          const digitsOfPrecision = 4;
+
+          for (let modelId = 0; modelId < data.length; modelId++) {
+            const tempX = series.x ? series.x[modelId] : undefined;
+            const tempY = series.y ? series.y[modelId] : undefined;
+            const lowerErrorX =
+              series?.error_x?.type === "data"
+                ? series.error_x.arrayminus?.[modelId]
+                : undefined;
+            const upperErrorX =
+              series?.error_x?.type === "data"
+                ? series.error_x.array?.[modelId]
+                : undefined;
+            const lowerErrorY =
+              series?.error_y?.type === "data"
+                ? series.error_y.arrayminus?.[modelId]
+                : undefined;
+            const upperErrorY =
+              series?.error_y?.type === "data"
+                ? series.error_y.array?.[modelId]
+                : undefined;
+
+            const x =
+              series.x && tempX !== undefined && typeof tempX == "number"
+                ? tempX
+                : undefined;
+            const y =
+              series.y && tempY !== undefined && typeof tempY == "number"
+                ? tempY
+                : undefined;
+
+            const xBounds =
+              lowerErrorX !== 0 &&
+              upperErrorX !== 0 &&
+              typeof lowerErrorX == "number" &&
+              typeof upperErrorX == "number" &&
+              x !== undefined
+                ? `[${(x - lowerErrorX).toFixed(digitsOfPrecision)}, ${(
+                    x + upperErrorX
+                  ).toFixed(digitsOfPrecision)}]`
+                : "";
+            const yBounds =
```

Comment:
At first glance, this appears to be the same as the `xBounds` calculation, swapping out `x` for `y`? If so, can it be refactored into a function?